src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -7,13 +7,220 @@
+    status = uct_rdmacm_create_dummy_cq_qp(cep->id, &cq, &qp);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    cep->cq                      = cq;
+    cep->qp                      = qp;
+
+    conn_param->qp_num           = qp->qp_num;
+    conn_param->private_data_len = sizeof(*hdr) + hdr->length;
+    return UCS_OK;
+}
+
+static ucs_status_t uct_rdamcm_cm_ep_client_init(uct_rdmacm_cm_ep_t *cep,
+                                                 const uct_ep_params_t *params)
+{
+    ucs_status_t status;
+
+    cep->wireup.client.connect_cb = params->sockaddr_connect_cb.client;
+
+    if (rdma_create_id(cep->cm->ev_ch, &cep->id, cep, RDMA_PS_TCP)) {
+        ucs_error("rdma_create_id() failed: %m");
+        status = UCS_ERR_IO_ERROR;
+        goto err;
+    }
+
+    /* rdma_resolve_addr needs to be called last in the ep_create flow to
+     * prevent a race where there are unitilialized fields used when the
+     * RDMA_CM_EVENT_ROUTE_RESOLVED event is already received in the the async
+     * thread. */
+    if (rdma_resolve_addr(cep->id, NULL, (struct sockaddr *)params->sockaddr->addr,
+                          1000/* TODO */)) {
+        ucs_error("rdma_resolve_addr() failed: %m");
+        status = UCS_ERR_IO_ERROR;
+        goto err_destroy_id;
+    }
+
+    return UCS_OK;
+
+err_destroy_id:
+    rdma_destroy_id(cep->id);
+err:
+    return status;
+}
+
+static ucs_status_t uct_rdamcm_cm_ep_server_init(uct_rdmacm_cm_ep_t *cep,
+                                                 const uct_ep_params_t *params)
 {
     return UCS_ERR_NOT_IMPLEMENTED;
 }
 
+ucs_status_t uct_rdmacm_cm_ep_disconnect(uct_ep_h ep, unsigned flags)
+{
+    uct_rdmacm_cm_ep_t *cep = ucs_derived_of(ep, uct_rdmacm_cm_ep_t);
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+
+    if (rdma_disconnect(cep->id)) {
+        ucs_error("rdmacm_cm ep %p (id=%p) failed to disconnect from peer %p",
+                  cep, cep->id,
+                  ucs_sockaddr_str(rdma_get_peer_addr(cep->id), ip_port_str,
+                                   UCS_SOCKADDR_STRING_LEN));
+        return UCS_ERR_IO_ERROR;
+    }
+
+    ucs_debug("rdmacm_cm ep %p (id=%p) disconnecting from peer :%s", cep, cep->id,
+              ucs_sockaddr_str(rdma_get_peer_addr(cep->id), ip_port_str,
+                               UCS_SOCKADDR_STRING_LEN));
+    return UCS_OK;
+}
+
+uct_base_iface_t dummy_iface = {
+    .super = {
+        .ops = {
+            .ep_pending_purge = (void *)ucs_empty_function_return_success,
+            .ep_disconnect    = uct_rdmacm_cm_ep_disconnect,
+            .ep_destroy       = UCS_CLASS_DELETE_FUNC_NAME(uct_rdmacm_cm_ep_t)
+        }
+    }
+};
+
+UCS_CLASS_INIT_FUNC(uct_rdmacm_cm_ep_t, const uct_ep_params_t *params)
+{
+    ucs_status_t status;
+
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_CM)) {
+        ucs_print("No UCT_EP_PARAM_FIELD_CM set: %zu", params->field_mask);
```

Comment:
"UCT_EP_PARAM_FIELD_CM is not set"

Reply:
it's a debug print that will be removed..
i will set it to ucs_debug after the issue is resolved.